lib/src/component_declaration/builder_helpers.dart

```diff
@@ -134,6 +134,84 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
+extension PropsToForward<T extends UiProps> on T {
+
+  /// Returns a copy of this instance's props excluding the keys found in [exclude].
+  ///
+  /// [exclude] should be a `Set` of PropsMixin `Type`s.
+  /// If [exclude] is not set it defaults to using the current instances Type.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// // within a functional component: `uiFunction<FooPropsMixin>`
+  /// // filter out the current components props when forwarding to Bar.
+  /// return (Bar()..addAll(props.getPropsToForward()))();
+  /// ```
+  /// OR
+  /// ```dart
+  /// // within a functional component that has multiple mixins on a Props class: `uiFunction<FooProps>`
+  /// // filter out the Components props when forwarding to Bar.
+  /// return (Bar()..addAll(props.getPropsToForward(exclude: {FooPropsMixin}))();
+  /// ```
+  ///
+  /// To only add DOM props, use the [domOnly] named argument.
+  ///
+  /// Related: `UiComponent2`'s `addUnconsumedProps`
+  Map getPropsToForward({Set<Type> exclude, bool domOnly = false}) {
+    return _propsToForward(exclude: exclude, domOnly: domOnly, propsToUpdate: {});
+  }
+
+  /// A utility function to be used with `modifyProps` to add props excluding the keys found in [exclude].
+  ///
+  /// [exclude] should be a `Set` of PropsMixin `Type`s.
+  /// If [exclude] is not set it defaults to using the current instances Type.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// // within a functional component: `uiFunction<FooPropsMixin>`
+  /// // filter out the current components props when forwarding to Bar.
+  /// return (Bar()..modifyProps(props.addPropsToForward()))();
+  /// ```
+  /// OR
+  /// ```dart
+  /// // within a functional component that has multiple mixins on a Props class: `uiFunction<FooProps>`
+  /// // filter out the Components props when forwarding to Bar.
+  /// return (Bar()..modifyProps(props.addPropsToForward(exclude: {FooPropsMixin}))();
+  /// ```
+  ///
+  /// To only add DOM props, use the [domOnly] named argument.
+  ///
+  /// Related: `UiComponent2`'s `addUnconsumedProps`
+  PropsModifier addPropsToForward({Set<Type> exclude, bool domOnly = false}) {
+    return (Map<dynamic, dynamic> props) {
+      _propsToForward(exclude: exclude, domOnly: domOnly, propsToUpdate: props);
+    };
+  }
+
+  Map _propsToForward({Set<Type> exclude, bool domOnly = false, Map propsToUpdate}) {
+    Iterable<PropsMeta> consumedProps = [];
+    try {
+      consumedProps = staticMeta.forMixins(exclude ?? {T}).toList();
+    } catch(_) {
+      // If [domOnly] is `true`, it is alright for the meta lookup to fail, otherwise throw the error.
+      assert(exclude == null && domOnly == true, ArgumentError('Could not find props meta for type $T.'
+        ' If this is not a props mixin, you need to specify its mixins as the second argument.  For example:'
+        '\n  ..addAll(props.getPropsToForward(exclude: {${T}Mixin})').message);
+    }
```

Comment:
Seems like we'd want to not swallow any non-`assert` errors here
```suggestion
      rethrow;
    }
```

Reply:
would this not cause it to throw the error even if the above assert did not trigger? it should probably be wrapped in a like if right?